### PR TITLE
ref(endpoints): Move endpoints to module

### PR DESCRIPTION
**Pull request by mgaeta**
These endpoints:
- ExternalTeamDetailsEndpoint
- ExternalTeamEndpoint
- ExternalUserDetailsEndpoint
- ExternalUserEndpoint
- ProjectCodeOwnersDetailsEndpoint
- ProjectCodeOwnersEndpoint
- ProjectCodeOwnersRequestEndpoint

are all related and behind the same feature flag. This PR moves them into their own module.

### CODE REVIEW

File: mypy.ini [hunk 1]
```diff
@@ -4,14 +4,9 @@ files = src/sentry/analytics/,
...
         src/sentry/api/bases/organization_events.py,
         src/sentry/api/bases/rule.py,
-        src/sentry/api/endpoints/external_team.py,
-        src/sentry/api/endpoints/external_team_details.py,
-        src/sentry/api/endpoints/external_user.py,
-        src/sentry/api/endpoints/external_user_details.py,
         src/sentry/api/endpoints/organization_events_stats.py,
         src/sentry/api/endpoints/organization_events_trace.py,
         src/sentry/api/endpoints/project_app_store_connect_credentials.py,
-        src/sentry/api/endpoints/project_codeowners.py,
```
Comment (leeandher) [inline mypy.ini:14]: Should the new module be added to mypi?

**Review by leeandher — COMMENTED** [review]
Just one question before approving 👍

File: mypy.ini (hunk 1 above)
Comment (mgaeta) [reply mypy.ini:14]: Yes that's in another PR 👍🏽

**Review by leeandher — APPROVED** [review]
Looks good to me, once CI is all fixed up 👌

**Pull request merged**